content/doc/book/system-administration/reverse-proxy-configuration-with-jenkins.adoc

```diff
@@ -0,0 +1,1411 @@
+      proxy_set_header   X-Forwarded-For   $proxy_add_x_forwarded_for;
+      proxy_set_header   X-Forwarded-Proto $scheme;
+      proxy_max_temp_file_size 0;
+
+      #this is the maximum upload size
+      client_max_body_size       10m;
+      client_body_buffer_size    128k;
+
+      proxy_connect_timeout      90;
+      proxy_send_timeout         90;
+      proxy_read_timeout         90;
+      proxy_buffering            off;
+      proxy_request_buffering    off; # Required for HTTP CLI commands
+      proxy_set_header Connection ""; # Clear for keepalive
+  }
+
+}
+----
+
+This assumes that you run Jenkins on port 8080.
+Remember to create the folder /var/log/nginx/jenkins.
+
+For this set up to work, the context path of Jenkins must be the same
+between your Nginx and Jenkins (that is, you can't run Jenkins on
+http://localhost:8081/ci and have it exposed at
+http://localhost:80/jenkins).
+
+Set the context path by modifying the hudson.xml configuration file and
+adding --prefix=/hudson (or similar) to the <arguments> entry.
+
+If you are having problems with some paths (eg folders) with *Blue
+Ocean*, you may need to add the following snippet to your proxy
+configuration:
+
+[source]
+----
+if ($request_uri ~* "/blue(/.*)") {
+    proxy_pass http://YOUR_SERVER_IP:YOUR_JENKINS_PORT/blue$1;
+    break;
+}
+----
+
+To give nginx permission to read Jenkins web root folder, add nginx to
+the Jenkins group:
+
+[source,sh]
+----
+useradd -aG jenkins nginx
+----
+
+If you are experiencing timeouts when attempting to run long CLI
+commands through a proxy in Jenkins, you can increase the
+`+proxy_read_timeout+` setting as necessary.
+Older versions of Jenkins may not respect the `+proxy_read_timeout+` setting.
+
+If you are experiencing the following error when attempting to run long
+CLI commands in Jenkins and Jenkins is running behind Nginx, it
+is probably due to Nginx timing out the CLI connection.
+You can increase the `+proxy_read_timeout+` setting as necessary so the command will
+complete successfully.
+
+[source]
+----
+WARNING: null
+hudson.cli.DiagnosedStreamCorruptionException
+Read back: 0x00 0x00 0x00 0x1e 0x07
+           'Started reverse-proxy-test #68'
+           0x00 0x00 0x00 0x01 0x07 0x0a
+Read ahead:
+Diagnosis problem:
+    java.io.IOException: Premature EOF
+        at sun.net.www.http.ChunkedInputStream.readAheadBlocking(ChunkedInputStream.java:565)
+        ...
+    at hudson.cli.FlightRecorderInputStream.analyzeCrash(FlightRecorderInputStream.java:82)
+    at hudson.cli.PlainCLIProtocol$EitherSide$Reader.run(PlainCLIProtocol.java:153)
+Caused by: java.io.IOException: Premature EOF
+    at sun.net.www.http.ChunkedInputStream.readAheadBlocking(ChunkedInputStream.java:565)
+    ...
+    at java.io.DataInputStream.readInt(DataInputStream.java:387)
+    at hudson.cli.PlainCLIProtocol$EitherSide$Reader.run(PlainCLIProtocol.java:111)
+----
+
+=== Forwarding With Iptables
```

Comment:
This should be at the same level as "Running behind Nginx" and "Running behind Apache" and should be included in the lists of links at the start of the file.  This is an alternate technique that uses the Linux networking components to transfer traffic from port 80 to port 8080 and from port 443 to 8443.
```suggestion
== Running Jenkins behind iptables
```

I'm not confident how the Linux project would choose to capitalize `iptables` when used in a title.  https://www.linode.com/docs/security/firewalls/control-network-traffic-with-iptables/ uses it with lower case 'i'.  https://help.ubuntu.com/community/IptablesHowTo uses it with capital 'I' in one title and lower case 'i' in another title (in the same article).  Arch Linux uses it as lower case 'i'.